src/servers/reducers/servers.ts

```diff
@@ -21,7 +21,7 @@ const serverWithId = (server: ServerWithId | ServerData): ServerWithId => {
     return server as ServerWithId;
   }
 
-  return assoc('id', uuid(), server);
+  return assoc('id', sha256(`${server.url}-${server.apiKey}`), server);
```

Comment:
I have three main concerns with this approach.

* The sha256 hash is sooooo loooong. It's quite overkill.
* The API key is "leaked" in the URL, which is not very secure. Even less if we try to address previous point by switch to a shorter but also less secure hashing algorithm.
* The hash is not unique, as a user could potentially set-up the same server more than once.

In order to address first two, I would instead use a different approach. Ideally, I would generate a slug of the title or the domain.

That way it's shorter, but also more human friendly and easier to spot which server are you opening. It also prevents the API key to be leaked in any way to the URL, even through a hash.

The third point is harder to address, but I think it's reasonable to append an index if the generated ID already exists, starting with 1. If it already exists, we increase it and try again.

When you create the servers from a servers.json file, the servers are processed in order, so the indexes should end up being deterministic.

In any other case, it will not be really a problem.

Reply:
Thanks for the review. I meant to keep working on this PR last night, but here we are 😛

I though about leaking info in the URL, yes. But: a) sha256 hasn't been broken yet; b) whoever gets this URL (say we send it as an HTTP referer by mistake) already has access to the server's URL, which would publish the `servers.json` - so it's kind of game over for pre-defined servers anyway.

I know I'm biased to thinking about pre-defined servers (since my use case is to have a single server, and I want using it to be straightforward) so I'll trust your point of view more than mine with this. If you think a slug + index would work best - I can give it a try 👍